### PR TITLE
0.42.0 Release

### PR DESCRIPTION
**Pull request by purva-thakre**
## Description

---

### License

- [ ] I license this contribution under the terms of the GNU GPL, version 3 and grant Unitary Fund the right to provide additional permissions as described in section 7 of the GNU GPL, version 3.

Before opening the PR, please ensure you have completed the following where appropriate.

- [ ] I added unit tests for new code.
- [ ] I used [type hints](https://www.python.org/dev/peps/pep-0484/) in function signatures.
- [ ] I used [Google-style](https://google.github.io/styleguide/pyguide.html#383-functions-and-methods) docstrings for functions.
- [ ] I [updated the documentation](../blob/main/docs/CONTRIBUTING_DOCS.md) where relevant.
- [ ] Added myself / the copyright holder to the AUTHORS file

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,48 @@
 
 ## Version 0.42.0
 
-In Development.
+([Full Changelog](https://github.com/unitaryfund/mitiq/compare/v0.41.0...v0.42.0))
+
+### Highlights
+
+🚀 Many thanks to **first time contributors** @gluonhiggs, @JMuff22 and @Shivansh20128!
```
Comment (natestemen) [inline CHANGELOG.md:10]: highlighting what they worked on would be great!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,48 @@
...
+
+💡 A new error-mitigation technique is on its way. Thanks to @Misty-W for the Probabilistic Error Amplification RFC!
```
Comment (natestemen) [inline CHANGELOG.md:12]: Can you add a link to the RFC? Let's say that it has been reviewed and accepted, and we will begin implementing the technique in Q1.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,48 @@
...
+
+🎏 Modular PEC functions are now available, courtesy of @natestemen!
```
Comment (natestemen) [inline CHANGELOG.md:14]: TIL 🎏!

Can we also add something like this:
````md
These functions allow a user to generate the intermediary sampled circuits and combine the results in a a two step process. E.g.

```py
from mitiq import pec

circuits = pec.intermediary_sampled_circuits(circuit, representations)

sampled_circuit_results = ...  # execute the circuits on a simulator/hardware/toothbrush/etc

pec_estimate = pec.combine_results(sampled_circuit_results, ...)
```
````

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,48 @@
...
+
+#### ✨ Enhancements
+
+- Fix converting Rxx and similar Qiskit gates (#2579) [@gluonhiggs]
+- Handle unsupported gates (#2585) [@gluonhiggs]
+- RFC for Probabilistic Error Amplification technique (#2550) [@Misty-W]
+- Address executor and observable incompatibility (#2514) [@bdg221]
+- Modularize PEC functionality (#2604) [@natestemen]
+-
```
Comment (natestemen) [inline CHANGELOG.md:23]:
```suggestion
```

File: CHANGELOG.md (hunk 3 above)
Comment (purva-thakre) [reply CHANGELOG.md:14]: > TIL 🎏!

Yeah! I got it from https://getemoji.com/#objects

**Comment by purva-thakre** [comment]
> here's a link to our newsletter

@natestemen Do I link the google doc?

**Review by natestemen — APPROVED** [review]
:shipit:

**Pull request merged**